v3/mem/mem_linux.go

```diff
@@ -73,86 +73,226 @@ func fillFromMeminfoWithContext(ctx context.Context) (*VirtualMemoryStat, *Virtu
 
-		t, err := strconv.ParseUint(value, 10, 64)
-		if err != nil {
-			return ret, retEx,err
-		}
 		switch key {
 		case "MemTotal":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			ret.Total = t * 1024
 		case "MemFree":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			ret.Free = t * 1024
 		case "MemAvailable":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			memavail = true
 			ret.Available = t * 1024
 		case "Buffers":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			ret.Buffers = t * 1024
 		case "Cached":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			ret.Cached = t * 1024
 		case "Active":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			ret.Active = t * 1024
 		case "Inactive":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			ret.Inactive = t * 1024
 		case "Active(anon)":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			retEx.ActiveAnon = t * 1024
 		case "Inactive(anon)":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			retEx.InactiveAnon = t * 1024
 		case "Active(file)":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			activeFile = true
 			retEx.ActiveFile = t * 1024
 		case "Inactive(file)":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			inactiveFile = true
 			retEx.InactiveFile = t * 1024
 		case "Unevictable":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			retEx.Unevictable = t * 1024
 		case "WriteBack":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			ret.WriteBack = t * 1024
 		case "WriteBackTmp":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			ret.WriteBackTmp = t * 1024
 		case "Dirty":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			ret.Dirty = t * 1024
 		case "Shmem":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			ret.Shared = t * 1024
 		case "Slab":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			ret.Slab = t * 1024
-		case "Sreclaimable":
+		case "SReclaimable":
+			t, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				return ret, retEx, err
+			}
 			sReclaimable = true
 			ret.Sreclaimable = t * 1024
-		case "Sunreclaim":
+		case "SUnreclaim":
```

Comment:
wow, didn't noticed, sorry. it is because replacing on v3.

Reply:
I found this regression when porting the v2 mock to v3. I plan (and already started) to add mocked tests to prevent regressions (and do a bit of test driven development for #881 and #990) for the cpu package on linux (parsing `/proc/cpuinfo`).

I'm not sure why you renamed these fields in v3 in fact.

Reply:
I renamed all by using sed, and i didn't noticed this also replaced. sorry